utils/model.py

```diff
@@ -66,10 +66,10 @@ def __init__(self, classes, load_path_scaler=None, load_path_PCA=None, load_path
         else:
             self.callbacks = []
 
-    def predict(self, X):
+    def predict(self, X, threshold_nn=0.5):
```

Comment:
I would remove that as an optional parameter and would make it compulsory.

Reply:
Hmmm Interesting Point